flamingo/grammar/assignment.h

```diff
@@ -87,17 +88,35 @@ static int parse_assignment(flamingo_t* flamingo, TSNode node) {
 		return -1;
 	}
 
+	if (rhs->kind != prev_type && (prev_type != FLAMINGO_VAL_KIND_NONE && rhs->kind != FLAMINGO_VAL_KIND_NONE)) {
+		return error(flamingo, "cannot assign %s to '%.*s' (%s)", val_type_str(rhs), (int) lhs_size, lhs, prev_type_str);
+	}
+
 	if (var != NULL) {
-		val_decref(val);
+		// Re-find variable to be safe against reallocs.
```

Comment:
which reallocs? I guess it could be possible for `parse_index` to realloc the vars list, in which case this may be indicative of a deeper problem that I'd have to think about a little more

Reply:
It's just a safety precaution in case parse_expr triggers anything that reallocates the variable storage, invalidating our var pointer.

Reply:
yup, so probably something to keep an eye on. Opened an issue: #16